pkg/common_controller/snapshot_controller.go

```diff
@@ -502,362 +645,270 @@ func (ctrl *csiSnapshotController) isVolumeBeingCreatedFromSnapshot(snapshot *cr
-	  	snapshotClassName := content.Spec.VolumeSnapshotClassName
-		if snapshotClassName != nil {
-			if snapshotClass, err := ctrl.classLister.Get(*snapshotClassName); err == nil {
-				// Resolve snapshotting secret credentials.
-				// No VolumeSnapshot is provided when resolving delete secret names, since the VolumeSnapshot may or may not exist at delete time.
-				snapshotterSecretRef, err := getSecretReference(snapshotClass.Parameters, content.Name, nil)
-				if err != nil {
-					return err
-				}
-				snapshotterCredentials, err = getCredentials(ctrl.client, snapshotterSecretRef)
-				if err != nil {
-					return err
-				}
+	// update snapshot status
+	if snapshotContent.Status != nil {
+		// Update snapshot status with creationTime
+		for i := 0; i < ctrl.createSnapshotContentRetryCount; i++ {
+			klog.V(5).Infof("syncUnreadySnapshot [%s]: trying to update snapshot status", utils.SnapshotKey(snapshotCopy))
+			_, err = ctrl.updateSnapshotStatus(snapshotCopy, snapshotContent)
+			if err == nil {
+				break
 			}
-		}*/
-
-	err := ctrl.handler.DeleteSnapshot(content, snapshotterCredentials)
-	if err != nil {
-		ctrl.eventRecorder.Event(content, v1.EventTypeWarning, "SnapshotDeleteError", "Failed to delete snapshot")
-		return fmt.Errorf("failed to delete snapshot %#v, err: %v", content.Name, err)
-	}
+			klog.V(4).Infof("failed to update snapshot %s status: %v", utils.SnapshotKey(snapshot), err)
+		}
 
-	err = ctrl.clientset.SnapshotV1beta1().VolumeSnapshotContents().Delete(content.Name, &metav1.DeleteOptions{})
-	if err != nil {
-		ctrl.eventRecorder.Event(content, v1.EventTypeWarning, "SnapshotContentObjectDeleteError", "Failed to delete snapshot content API object")
-		return fmt.Errorf("failed to delete VolumeSnapshotContent %s from API server: %q", content.Name, err)
+		if err != nil {
+			// update snapshot status failed
+			ctrl.updateSnapshotErrorStatusWithEvent(snapshotCopy, v1.EventTypeWarning, "SnapshotStatusUpdateFailed", fmt.Sprintf("Snapshot status update failed, %v", err))
+			return nil, err
+		}
 	}
 
-	return nil
+	klog.V(5).Infof("bindandUpdateVolumeSnapshot for snapshot completed [%#v]", snapshotCopy)
+	return snapshotCopy, nil
 }
 
-// TODO(xiangqian) This is a temp implementation to make sure the test cases pass
-// updateSnapshotContentStatus updates status of a content object
-func (ctrl *csiSnapshotController) updateSnapshotContentStatus(
-	content *crdv1.VolumeSnapshotContent,
-	snapshotHandle string,
-	readyToUse bool,
-	createdAt int64,
-	size int64) error {
-	var newStatus *crdv1.VolumeSnapshotContentStatus
-	updated := false
+// UpdateSnapshotStatus updates snapshot status based on content status
+func (ctrl *csiSnapshotCommonController) updateSnapshotStatus(snapshot *crdv1.VolumeSnapshot, content *crdv1.VolumeSnapshotContent) (*crdv1.VolumeSnapshot, error) {
+	klog.V(5).Infof("updateSnapshotStatus[%s]", utils.SnapshotKey(snapshot))
 	if content.Status == nil {
```

Comment:
should BoundContentName be updated even if the status of content is nil?

Reply:
That's my question too.